packages/stryker/test/unit/mutators/BooleanSubstitutionMutatorSpec.ts

```diff
@@ -0,0 +1,56 @@
+import { expect } from 'chai';
+import * as estree from 'estree';
+import { Identified } from 'stryker-api/mutant';
+import BooleanSubstitutionMutator from '../../../src/mutators/BooleanSubstitutionMutator';
+import { parse } from '../../../src/utils/parserUtils';
+import { copy } from '../../../src/utils/objectUtils';
+
+describe('BooleanSubstitutionMutator', () => {
+  let sut: BooleanSubstitutionMutator;
+
+  beforeEach(() => sut = new BooleanSubstitutionMutator());
+
+  it('should mutate when supplied a expression with !', () => {
+    // Arrange
+    const program = parse(`!a.a()`);
+    const nodeUnaryExpression = ((program.body[0] as estree.ExpressionStatement).expression as estree.UnaryExpression & Identified);
+
+    // Act
+    const result = sut.applyMutations(nodeUnaryExpression, copy)[0] as estree.Expression & Identified;
+
+    // Assert
+    expect(result).to.be.ok;
+    expect(result.type).to.be.eq(nodeUnaryExpression.argument.type);
```

Comment:
It will most probably not be a Syntax.UnaryExpression. In the tree I move the Expression(any Expression) in the UnaryExpression (argument) to the position of the UnaryExpression.